### PR TITLE
added new options: api, start-on-load

### PR DESCRIPTION
**Pull request by yaroslawww**
Add support for programmatic initialise (publish).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -104,6 +112,14 @@ By default smartbanner would reappear if site path changes. Following example wo
...
 ```
 
+### Set "smartbanner" as global variable
+
+By default smartbanner not support manual construction. You can enable it and use "window.smartbanner" variable
+
+```html
+<meta name="smartbanner:api" content="true">
+```
+
```
Comment (ain) [inline README.md]: Whenever API is used, publishing should also be a responsibility of a developer which would make `start-on-load` obsolete. I'd support single option here, the API.

File: src/index.js [hunk 2]
```diff
@@ -4,5 +4,10 @@ let smartbanner;
 
 window.addEventListener('load', function() {
   smartbanner = new SmartBanner();
-  smartbanner.publish();
+  if(smartbanner.apiEnabled) {
+    window.smartbanner = smartbanner;
+  }
+  if(smartbanner.startOnLoad) {
+    smartbanner.publish();
+  }
```
Comment (ain) [inline src/index.js]: See comment above.

File: src/smartbanner.js [hunk 3]
```diff
@@ -135,6 +135,13 @@ export default class SmartBanner {
     return height !== undefined ? height : 0;
   }
 
+  get startOnLoad() {
+    if(this.options.startOnLoad === undefined) {
+      this.options.startOnLoad = 'true';
+    }
+    return this.options.startOnLoad === 'true';
+  }
```
Comment (ain) [inline src/smartbanner.js]: See comment above.

**Comment by yaroslawww** [comment]
Hi, I updated code and added test. Can you check please?

**Comment by alcferreira** [comment]
Hey @ain, how is it going?
Do you have any ideia if this PR is going to be merged soon? 
Do you need something else here?

File: README.md (hunk 1 above)
Comment (yaroslawww) [reply README.md]: Removed obsolete `start-on-load`

File: src/index.js (hunk 2 above)
Comment (yaroslawww) [reply src/index.js]: Removed obsolete `start-on-load`

File: test/spec/smartbanner_spec.js [hunk 4]
```diff
@@ -87,6 +87,28 @@ describe('SmartBanner', function() {
     </body>
   </html>`;
 
+  const HTML_API = `<!doctype html>
```
Comment (yaroslawww) [inline test/spec/smartbanner_spec.js]: Added test for  `smartbanner:api`

**Comment by ain** [comment]
Thanks for the work here guys, this will be reviewed next targeting the release of `1.12.0` (given that it introduces no breaking change).

File: test/spec/smartbanner_spec.js [hunk 5]
```diff
@@ -708,4 +730,34 @@ describe('SmartBanner', function() {
...
+
+    before(function() {
+      global.window = new JSDOM(HTML_API, {userAgent: USER_AGENT_IPHONE_IOS9}).window;
+      global.document = window.document;
+      global.getComputedStyle = window.getComputedStyle;
+      smartbanner = new SmartBanner();
+    });
+
+    it('expected to smartbanner.apiEnabled is true ', function() {
+      expect(smartbanner.apiEnabled).to.be.true;
```
Comment (ain) [inline test/spec/smartbanner_spec.js]: This test here is more about the options test that is already covered in option parser above. I'd replace it with a functional test and check if

- global object is added to `window` if API is turned on
- nothing is added to DOM if API is turned on
- component is added to DOM once `publish` is called 
- `smartbanner.view` event is only fired when API is used and `publish` gets called

File: test/spec/optionparser_spec.js [hunk 6]
```diff
@@ -108,6 +109,10 @@ describe('OptionParser', function() {
       it('expected to parse hide path option', function() {
         expect(options.hidePath).to.eql('/smartbanner');
       });
+
+      it('expected to parse api option', function() {
+        expect(options.api).to.eql('true');
+      });
```
Comment (ain) [inline test/spec/optionparser_spec.js]: 👍

File: src/index.js [hunk 7]
```diff
@@ -4,5 +4,9 @@ let smartbanner;
 
 window.addEventListener('load', function() {
   smartbanner = new SmartBanner();
-  smartbanner.publish();
+  if(smartbanner.apiEnabled) {
```
Comment (ain) [inline src/index.js]:
```suggestion
  if (smartbanner.apiEnabled) {
```

File: README.md [hunk 8]
```diff
@@ -104,6 +104,20 @@ By default smartbanner would reappear if site path changes. Following example wo
 <meta name="smartbanner:hide-path" content="/">
 ```
 
+### Set "smartbanner" as global variable
+
+By default smartbanner not support manual construction. You can enable it and use "window.smartbanner" variable
```
Comment (ain) [inline README.md]:
```suggestion
By default smartbanner is added to DOM automatically. You can disable it with
```

File: README.md [hunk 9]
```diff
@@ -104,6 +104,20 @@ By default smartbanner would reappear if site path changes. Following example wo
...
+
+```html
+<meta name="smartbanner:api" content="true">
+```
+
+**Note:** By default smartbanner initialise on window load. But when used option *api* you should initialise smartbanner manually
```
Comment (ain) [inline README.md]:
```suggestion
and add smartbanner to DOM manually:
```

File: README.md [hunk 10]
```diff
@@ -104,6 +104,20 @@ By default smartbanner would reappear if site path changes. Following example wo
...
+By default smartbanner not support manual construction. You can enable it and use "window.smartbanner" variable
+
+```html
+<meta name="smartbanner:api" content="true">
+```
+
+**Note:** By default smartbanner initialise on window load. But when used option *api* you should initialise smartbanner manually 
+
+```js
+smartbanner.publish()
```
Comment (ain) [inline README.md]:
```suggestion
smartbanner.publish();
```

File: README.md [hunk 11]
```diff
@@ -104,6 +104,20 @@ By default smartbanner would reappear if site path changes. Following example wo
 <meta name="smartbanner:hide-path" content="/">
 ```
 
+### Set "smartbanner" as global variable
```
Comment (ain) [inline README.md]:
```suggestion
### smartbanner API use
```

File: test/spec/smartbanner_spec.js (hunk 5 above)
Comment (yaroslawww) [reply test/spec/smartbanner_spec.js]: Unfortunately, I have no experience in writing functional tests.

**Comment by sebastianhelbig** [comment]
Any updates here?

**Comment by ain** [comment]
Not at this point, no. I'd love someone to pick it up and complete the tests.

**Comment by ain** [comment]
Work continued in #124

**Comment by ain** [comment]
Thanks again for your work @yaroslawww, I'll get it released soon. First tests were successful.

**Comment by yaroslawww** [comment]
@ain  Thank you for continuing to work and help with the tests!

**Pull request closed without merging**